### PR TITLE
Use systemd-255 macros in systemd-bootstrap

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

There is currently no quick path to upgrading systemd-bootstrap, but we need to align to the macros in the new systemd-255. Packages that use macros like `%systemd_requires` or `%systemd_post` use them statically, i.e., the macros are evaluated at `rpmbuild` time and encoded into the `.rpm`. This means that any package using `systemd-bootstrap-rpm-macros` to build would be left using outdated macros.

Until we can get `systemd-bootstrap` aligned with `systemd` we should at least make sure the macros and associated components are updated to match.

###### Change Log  

- Port systemd-255's macros to systemd-250

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local builds
- Full build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516187&view=results

### CODE REVIEW

File: SPECS/systemd/systemd.signatures.json [hunk 1]
```diff
@@ -5,10 +5,7 @@
   "10-oomd-per-slice-defaults.conf": "3f0d6882312affecddfdc5a204a44c34b90c573839f2a4850a26b44f528520bd",
   "10-timeout-abort.conf": "ae6b234f92bc22f1201a7572b59b454c9809f33c80d13f361b9674e1801acc37",
   "20-yama-ptrace.conf": "f7881466bff200865ec2c6b5f989ed35855ac45420a55ae5ace805f9e5111828",
-  "26494.patch": "1242a0db037a2c8e5de3c8b676aad63caaae60d674227ec58d1ea11f085a996b",
```
Comment (christopherco) [inline SPECS/systemd/systemd.signatures.json:8]: Curious why these diffs are present

**Review by christopherco — COMMENTED** [review]
Is there instructions posted somewhere on how to test for these macros to stay in sync?

File: SPECS/systemd/systemd.signatures.json (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/systemd/systemd.signatures.json:8]: These files either are not present, or are .patch files which are handled differently. I ran the tools with the auto hash update turned on so they got cleaned up. I can revert if desired.

**Comment by dmcilvaney** [comment]
> Is there instructions posted somewhere on how to test for these macros to stay in sync?

I can add a readme, but an almost entirely manual process unfortunately.

File: SPECS/systemd-bootstrap/systemd-bootstrap.spec [hunk 2]
```diff
@@ -26,6 +33,21 @@ Patch4:         CVE-2022-45873.patch
...
+#    3. Personal flow to create patch:
+#         cd BUILD/systemd-stable-250 && git init && git add -A && git commit -m "init"
+#         cp ../systemd-stable-255/src/rpm/macros.systemd.in src/rpm/macros.systemd.in
+#         # Adjust changes as needed (LIBEXECDIR->ROOTLIBEXECDIR, define USER_TMPFILES_DIR), possibly split each change into its own commit
+#         git add -A && git commit -m "use-255-macros"
+#         # Use -1 for a single commit, or -3 etc. for more
+#         git format-patch -1 --stdout > use-255-macros.patch
+#    4. Update patch, then rebuild to validate changes
+#    5. Repeat from 2. as needed until it builds
+#    6. Build both systemd and systemd-bootstrap, validate the contents of systemd-rpm-macros and system-bootstrap-rpm-macros are idential
```
Comment (dmcilvaney) [inline SPECS/systemd-bootstrap/systemd-bootstrap.spec:49]:
```suggestion
# Directions for refreshing systemd macros:
#    1. Update Source21->24 from main systemd directory, currently they are taken un-modified.
#    2. Take systemd and systemd-bootstrap .src.rpms, install and run rpmbuild -bp --nodeps --noclean <srpm>.
#    3. Create a patch:
#         cd BUILD/systemd-stable-250 && git init && git add -A && git commit -m "init"
#         cp ../systemd-stable-255/src/rpm/macros.systemd.in src/rpm/macros.systemd.in
#         # Adjust changes as needed (LIBEXECDIR->ROOTLIBEXECDIR, define USER_TMPFILES_DIR), ideally split each change into its own commit for ease of review.
#         git add -A && git commit -m "use-255-macros"
#         # Use -1 for a single commit, or -3 etc. for more
#         git format-patch -1 --stdout > use-255-macros.patch
#    4. Update patch, then rebuild to validate changes
#    5. Repeat from 2. as needed until it builds
#    6. Build both systemd and systemd-bootstrap, validate the contents of systemd-rpm-macros and system-bootstrap-rpm-macros are identical
```

File: SPECS/systemd/systemd.signatures.json (hunk 1 above)
Comment (christopherco) [reply SPECS/systemd/systemd.signatures.json:8]: It is fine to keep. Thanks!

**Review by ddstreetmicrosoft — APPROVED** [review]
lgtm

**Review by anphel31 — APPROVED** [review]

**Pull request merged**